applications/OptimizationApplication/custom_utilities/container_variable_data/container_variable_data.h

```diff
@@ -0,0 +1,241 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main author:     Suneth Warnakulasuriya
+//
+
+#pragma once
+
+// System includes
+#include <cmath>
+#include <string>
+#include <variant>
+#include <vector>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+
+// Application includes
+#include "custom_utilities/container_variable_data/expressions.h"
+
+namespace Kratos {
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @brief Container variable data holder
+ *
+ * Objects of this class is used to hold any type of data in a container
+ * of TContainerType. TContainerType can be nodes, conditions or elements.
```

Comment:
```suggestion
 * of TContainerType. TContainerType can be a container of nodes, conditions or elements.
```

Also, template params are usually described in `@tparam` annotation blocks. For example:
```cpp
@tparam A container of nodes, conditions or elements.
```